.github/workflows/pr.yaml

```diff
@@ -82,13 +98,37 @@ jobs:
       - name: Await backend to be ready
         run: until kubectl get capp platform-backend -n platform-backend-system -o jsonpath='{.status.knativeObjectStatus.conditions[?(@.type=="Ready")].status}' | grep -q "True"; do echo "Waiting for Capp to be ready..."; sleep 5; done; echo "Capp is ready"
 
+      - name: Set Kubernetes context
+        uses: Azure/k8s-set-context@v4
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.BACKEND_CI_CLUSTER_KUBECONFIG }}
+
       - name: Run e2e tests
         run: make test-e2e
 
+      - name: Set Kubernetes context
+        uses: Azure/k8s-set-context@v4
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.BACKEND_CI_HUB_CLUSTER_KUBECONFIG }}
+        if: always()
+
       - name: Undeploy backend
         run: make undeploy
         if: always()
 
+      - name: Cleanup hub
+        run: make cleanup-hub
+        if: always()
```

Comment:
The CI can potentially fail at any stage, so make sure to switch to the correct kube context before the cleanup

Reply:
Since they have `if: always()` shouldn't they run?

Reply:
Nevermind I missed some lines before